contributing-device-support.md

```diff
@@ -0,0 +1,126 @@
+
+    The resin-image.bbappend file shall define the following variables:
+
+    - IMAGE_FSTYPES_`<yocto-machine-name>`
+
+        This variable is used to declare the type of the produced image (it can be ext3, ext4, resin-sdcard etc. The usual type for a board that can boot from SD card, USB, is "resin-sdcard").
+
+    - RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>`
+
+        this allows adding files from the the build's deploy directory into the vfat formatted boot partition (can be used to add bootloader config files, first stage bootloader,
+        initramfs etc.).
+
+        It is a space separated list of items with the following format: *FilenameRelativeToDeployDir:FilenameOnTheTarget*. If *FilenameOnTheTarget* is omitted then the *FilenameRelativeToDeployDir* will be used.
+
+        For example to have the Intel NUC bzImage-intel-corei7-64.bin copied from deploy directory over to the boot partition, renamed to vmlinuz:
+        ```sh
+        RESIN_BOOT_PARTITION_FILES_nuc = "bzImage-intel-corei7-64.bin:vmlinuz"
+        ```
+
+    The resin-image-flasher.bbappend file shall define the following variables:
+
+    - IMAGE_FSTYPES_`<yocto-machine-name>` (see above)
+
+    - RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>` (see above).
+        For example, if the board uses different bootloader configuration files for booting from SD/USB and internal storage (see below the use of
+        INTERNAL_DEVICE_BOOTLOADER_CONFIG variable), then make sure these files end up in the boot partition (i.e. they should be listed in this RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>` variable)
+
+- recipes-kernel/linux directory: shall contain a .bbappend to the kernel recipe used by the respective board. This kernel .bbappend must "inherit kernel-resin" in order to add the necessary kernel configs for using with resin.io
+
+- recipes-support/resin-init directory - shall contain a resin-init-flasher.bbappend file if you intend to install resin.io to internal storage and hence use the flasher image. This shall define the following variables:
+
+    - INTERNAL_DEVICE_KERNEL_`<yocto-machine-name>`
+
+        This variable is used to identify the internal storage where resin.io will be written to.
+
+    - INTERNAL_DEVICE_BOOTLOADER_CONFIG_`<yocto-machine-name>`
+
+        This variable is used to specify the filename of the bootloader configuration file the board uses when booting from internal media (must be the same with the *FilenameOnTheTarget*
+        of the bootloader config file used in the RESIN_BOOT_PARTITION_FILES_`<yocto-machine-name>` variable from recipes-core/images/resin-image-flasher.bbappend);
+
+    - INTERNAL_DEVICE_BOOTLOADER_CONFIG_PATH_`<yocto-machine-name>`
+
+        This variable is used to specify the relative path (including filename) to /boot/ (this is currently partition 1) where INTERNAL_DEVICE_BOOTLOADER_CONFIG_`<yocto-machine-name>` will be copied to.
```

Comment:
We don't mount boot partition to /boot anymore. Let's not specify the mount-point but only the partition label.